File: programmers/lv2/124나라의숫자.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(int n) {
  string answer = "";
  int nameum = -1;
  while (n != 0) {
    nameum = n % 3;
    n /= 3;
    if (nameum == 0) {
      answer.append("4");
      n -= 1;
    }
    else if (nameum == 1)
      answer.append("1");
    else if (nameum == 2)
      answer.append("2");
  }
  reverse(answer.begin(), answer.end());
  return answer;
}

File: programmers/lv2/124나라의숫자.py
def solution(n):
    answer = []
    while(n > 0):
        restNum = n % 3
        n = n // 3
        if restNum == 0:
            answer.append('4')
            n -= 1
        elif restNum == 1:
            answer.append('1')
        elif restNum == 2:
            answer.append('2')
    answer.reverse()
    return ''.join(answer)